tools/mason/MasonPublish.chpl

```diff
@@ -722,3 +766,46 @@ private proc falseIfRemotePath() {
   }
   return true;
 }
+
+/* git tag version formatting */
+proc gitTagVersionCheck(projectHome: string) throws {
+  var tags = runCommand("git tag -l", true);
+  var allTags = tags.split("\n");
+  const toParse = open(projectHome + "/Mason.toml", iomode.r);
+  const tomlFile = owned.create(parseToml(toParse));
+  var version = "v" + tomlFile["brick"]!["version"]!.s;
+  for tag in allTags {
+    if tag == version {
+      return (true, allTags, version);
+    }
+  }
+  return (false, allTags, version);
+}
+
+/* make sure directory created is same as that of package 
+   name in manifest file */
+proc namespaceCollisionCheck(projectHome: string) throws {
+  var directoryName = basename(projectHome);
+  const toParse = open(projectHome + "/Mason.toml", iomode.r);
+  const tomlFile = owned.create(parseToml(toParse));
+  var packageName = tomlFile["brick"]!["name"]!.s;
+  if packageName == directoryName then return true;
+  else return false;
+}
+
+/* Mason toml file formatting */
+proc masonTomlFileCheck(projectHome: string) {
+  const toParse = open(projectHome + "/Mason.toml", iomode.r);
+  const tomlFile = owned.create(parseToml(toParse));
+  var missingFields : list(string);
+  var name, chplVersion, version, source, author, license = false;
+  if tomlFile.pathExists("brick.name") then name = true; else missingFields.append('name');
```

Comment:
We could wrap this in a helper if we wanted to clean it up slightly, but it's fine as is too, e.g.

```chpl
const name = fieldExists('name', missingFields);
const version = fieldExists('version', missingFields);
// ...
```

Reply:
I think the current code is quite straightforward, but I could refactor if you really emphasize on it.

Reply:
Yeah, I think it's fine as is.